studio.py

```diff
@@ -14,14 +22,211 @@
+            self.generate_manifest_file(recording)
+
+    def parse_sites_json(self):
+        print("Parsing sites json")
+        recordings = []
+        if self.sites is not None:
+            with open(self.sites, "r") as sites:
+                recordings_json = json.loads(sites.read())
+                self.information["app"] = self.app.lower()
+            if self.app in ["Firefox", "Chrome"]:
+                self.information["os_name"] = platform.system()
+                self.information["os_version"] = platform.release()
+                platform_name = platform.system().lower()
+
+            elif self.app in ["GeckoViewExample", "Fenix", "Fennec", "Refbrow"]:
+
+                device = ADBAndroid()
+                self.information["device_name"] = device.shell_output(
+                    "getprop ro.product.model"
+                )
+                self.information["os_name"] = device.shell_output(
+                    "getprop ro.build.user"
+                )
+                self.information["os_version"] = device.shell_output(
+                    "getprop ro.build.version.release"
+                )
+                if self.information["device_name"] == "Pixel 2":
+                    platform_name = "pixel2"
+                elif self.information["device_name"] == "Moto G (5)":
+                    platform_name = "motog5"
+
+            for recording in recordings_json:
+                label = recording.get("label", None)
+                if label:
+                    name = "-".join(
+                        [
+                            self.proxy,
+                            platform_name,
+                            self.app.lower(),
+                            recording["domain"],
+                            label,
+                        ]
+                    )
+                else:
+                    name = "-".join(
+                        [
+                            self.proxy,
+                            platform_name,
+                            self.app.lower(),
+                            recording["domain"],
+                        ]
+                    )
+                recording["path"] = os.path.join(self.path, name, name)
+                recording["name"] = name
+
+                recording["recording_path"] = "%s.mp" % recording["path"]
+                recording["json_path"] = "%s.json" % recording["path"]
+                recording["screen_shot_path"] = "%s.png" % recording["path"]
+                recording["zip_path"] = os.path.join(
+                    self.path, "%s.zip" % recording["name"]
+                )
+                recording["manifest_path"] = os.path.join(
+                    self.path, "%s.manifest" % recording["name"]
+                )
+
+                recordings.append(recording)
+        else:
+            raise Exception("No site file found!!!")
+
+        return recordings
+
+    def update_json_information(self, recording):
+        time.sleep(3)
```

Comment:
Why sleep here?

Reply:
Just to make sure that the proxy is closed and we can start edit the file.

This file is generated in after we issue the proxy stop command